R/get_panel.R

```diff
@@ -1,19 +1,59 @@
 #' Retrieve the panel of a plot
 #'
-#' This function extracts just the main panel from a ggplot. It only works for plots with exactly
-#' one panel (i.e., plots that are not faceted).
+#' `get_panel()` extracts just the main panel from a ggplot or a specified panel
+#' in a facetted plot. `get_panel_component()` extracts components from the
+#' panel, such as geoms.
 #'
 #' @param plot A ggplot or gtable from which to retrieve the panel
-#' @return A gtable object holding just the panel
+#' @param panel An integer indicating which panel to pull. ggplot orders panels
+#'   column-wise, so this is in order from the top left down.
+#' @param return_all If there is more than one panel, should all be returned
+#'   as a list? Default is `FALSE`.
+#' @param pattern the name of the component
+#' @return A gtable object holding the panel(s) or a grob of the component
+#' @examples
+#' library(ggplot2)
+#'
+#' p <- ggplot(mpg, aes(displ, cty)) + geom_point()
+#' plot_panel <- get_panel(p)
+#' ggdraw(plot_panel)
+#'
+#' ggdraw(get_panel_component(plot_panel, "geom_point"))
+#'
+#' @export
+get_panel <- function(plot, panel = NULL, return_all = FALSE) {
+  plot <- as_gtable(plot)
+  panel_locs <- grepl("panel", plot_component_names(plot))
+  panels <- plot_components(plot)[panel_locs]
+  if (length(panels) > 1 & is.null(panel) & !return_all)
```

Comment:
Please use `&&` in `if` statements.

Reply:
Oy, one day I'll remember `&&`/`||` is the right set for `if`. Maybe...